### PR TITLE
feat: add sway support

### PR DESCRIPTION
**Pull request by yehia67**
Add [Sway](https://fuellabs.github.io/sway/v0.51.1/book/index.html) Support.

### CODE REVIEW

**Comment by clason** [comment]
Textobjects don't live in this repository; pkease remove.

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1916,6 +1916,15 @@ list.svelte = {
   maintainers = { "@amaanq" },
 }
 
+list.sway = {
+  install_info = {
+    url = "https://github.com/FuelLabs/tree-sitter-sway",
+    files = { "src/parser.c", "src/scanner.c" },
+    requires_generate_from_grammar = true,
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:1923]:
```suggestion
```
Not needed since the repo contains `parser.c`.

File: queries/sway/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,362 @@
+; -------
+; Tree-Sitter doesn't allow overrides in regards to captures,
+; though it is possible to affect the child node of a captured
+; node. Thus, the approach here is to flip the order so that
+; overrides are unnecessary.
```
Comment (clason) [inline queries/sway/highlights.scm:5]: This is not true (in Neovim and the upcoming tree-sitter release).

**Comment by clason** [comment]
Please write your queries according to CONTRIBUTING.md; you **cannot** just copy queries from other editors.

File: queries/sway/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+; -------
+; -------
+; Types
+; -------
+; ---
+; Primitives
+; ---
+(escape_sequence) @string.escape
```
Comment (sdankel) [inline queries/sway/highlights.scm:13]: I believe this should be a copy of this file: https://github.com/FuelLabs/tree-sitter-sway/blob/master/queries/highlights.scm

Comment (clason) [reply queries/sway/highlights.scm:13]: Not a copy (in general); we have our own guidelines that queries **must** follow. Can be used as a base, though.

Comment (clason) [reply queries/sway/highlights.scm:13]: Yeah, these are Helix captures. Ours are somewhat different (simpler/flatter).

**Comment by clason** [comment]
@yehia67 are you actually willing to make the necessary changes to the queries (and keep maintaining them)?

**Comment by sdankel** [comment]
> @sdankel have you actually volunteered to maintain the queries _in this repository_?

Yes. @yehia67 volunteered to take the task of adding support for Sway here, and I will maintain it.

**Comment by clason** [comment]
Maybe you can help fixing the queries then?

**Comment by clason** [comment]
@sdankel

**Comment by clason** [comment]
I'm sorry, but this seems to be going nowhere. I am closing this PR so nobody else feels blocked from making a PR with proper queries.

**Pull request closed without merging**